src/state/apis/zapper/zapperApi.ts

```diff
@@ -79,37 +82,66 @@ export const zapperApi = createApi({
           },
         })
 
-        const data = zapperV2AppTokensData.map(appTokenData =>
-          toAssetId({
-            chainId: zapperNetworkToChainId(appTokenData.network)!,
-            assetNamespace: 'erc20', // TODO: bep20
-            assetReference: appTokenData.address,
-          }),
+        const data = zapperV2AppTokensData.reduce<GetZapperAppBalancesOutput>(
+          (acc, appTokenData) => {
+            const assetId = toAssetId({
+              chainId: zapperNetworkToChainId(appTokenData.network),
```

Comment:
would strongly prefer if we check if `appTokenData.network` is undefined and `return acc` along with reverting the return type sig to `ChainId | undefined` above

Reply:
cdc6d37b9016ba65b7cb12e6717e1a4c633cad23